api/incubator/src/main/java/io/opentelemetry/api/incubator/propagation/CaseInsensitiveMap.java

```diff
@@ -15,17 +15,25 @@ class CaseInsensitiveMap extends HashMap<String, String> {
   private static final long serialVersionUID = -4202518750189126871L;
 
   CaseInsensitiveMap(Map<String, String> carrier) {
-    super(carrier);
+    super();
+    // 20240617 fix bug
+    // if someone use Constructor to build this Map, this Map's key
+    // with be not lowerCase
+    carrier.forEach(this::put);
   }
 
   @Override
   public String put(String key, String value) {
-    return super.put(key.toLowerCase(Locale.ROOT), value);
+    return super.put(getKeyLowerCase(key), value);
+  }
+
+  protected String getKeyLowerCase(String key) {
```

Comment:
can this be `private`?

Reply:
At first, I plan to define it as private, so that unit test cannot access it，private func maybe  use reflection testing.

Reply:
if you need it for testing, make it default (package) access, rather than protected, and annotate it as `@VisibleForTesting`